### PR TITLE
Add depth to chart

### LINKED ISSUE #187: If a spotter is installed, the chart should show both the surface temps as well as the temps at depth (designs attached)
Screen Shot 2020-09-11 at 9 09 40 AM

### PR DESCRIPTION
**Pull request by SimplyBallistic**
Closes #187 
QOL:
- Added chart.js types
- Made React extend eslint config, ignored currently outstanding eslint issues.
- Reduced duplication

Screen Shot 2020-10-24 at 6 45 14 PM

### CODE REVIEW

File: packages/website/src/common/Chart/ChartWithTooltip.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ function ChartWithTooltip({
 
   const customTooltip = (ref: React.RefObject<Line>) => (tooltipModel: any) => {
     const chart = ref.current;
-    if (!chart) {
+    if (!chart || !chart.chartInstance.canvas) {
```
Comment (ericboucher) [inline packages/website/src/common/Chart/ChartWithTooltip.tsx:48]: Maybe you can just test for the second one with optional chaining?

File: packages/website/src/common/Chart/plugins/backgroundPlugin.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 const { Chart } = require("react-chartjs-2");
-
+/* eslint-disable fp/no-mutation,no-param-reassign */
```
Comment (ericboucher) [inline packages/website/src/common/Chart/plugins/backgroundPlugin.ts:2]: let's keep these disables at a line level when possible

File: packages/website/src/common/Chart/utils.ts [hunk 3]
```diff
@@ -111,3 +118,66 @@ export function useProcessedChartData(
...
   return { sortedDailyData, ...axisLimits, ...datasets };
 }
+
+export const createChartData = (
+  labels: string[],
+  tempWithSurvey: (number | null)[],
+  surfaceTemps: number[],
+  bottomTemps: number[],
+  fill: boolean
+  // TODO - use pointRadius to display survey dates
```
Comment (ericboucher) [inline packages/website/src/common/Chart/utils.ts:128]: Let's remove this TODO since it has been implemented

**Comment by ericboucher** [comment]
Looks great, is there any way to move the survey points to the bottom temp when possible?

See https://github.com/aqualinkorg/aqualink-app/issues/287

**Comment by SimplyBallistic** [comment]
@ericboucher Think it should do that now, make sure to test with prod - dev doesn't have any survey points taken when bottom temps data is introduced.

**Comment by SimplyBallistic** [comment]
https://github.com/facebook/create-react-app/issues/8320

**Review by ericboucher — APPROVED** [review]
Let's investigate the tooltip reactivity a little but it looks great otherwise, nice work!

**Pull request merged**